### PR TITLE
fix canary so it wont shit pants

### PR DESCRIPTION
**Pull request by pieckenst**

### CODE REVIEW

**Comment by pieckenst** [comment]
Damn thing just merge

File: listener/music/music.py [hunk 1]
```diff
@@ -1,240 +1,734 @@
...
+        """Update the player controller."""
+        player: Player = self.bot.wavelink.get_player(guild_id=ctx.guild.id, cls=Player, context=ctx)
+
+        if not player.is_connected:
+            return
+
+        await player.invoke_controller()
+
+    @commands.command(aliases=['swap'])
+    async def swap_dj(self, ctx: commands.Context, *, member: discord.Member = None):
```
[the remark this answers is not in the document]
Comment (pieckenst) [reply listener/music/music.py:700]: @muse-dev ignore

**Comment by pieckenst** [comment]
Merge for fuck sake

File: listener/wallpapers.py [hunk 2]
```diff
@@ -5,31 +5,32 @@
...
 
     @wallpaper.command()
-    async def nature(self, ctx):
+    async def nature(self, ctx: SlashContext):
         imgnat = random.choice(desNature.images)
         embednat = discord.Embed(title="Nature Wallpapers", color=0x00ff00, url=imgnat)
         embednat.set_image(url=imgnat)
         await ctx.send(embed=embednat)
     
     @wallpaper.command()
```
[the remark this answers is not in the document]
Comment (pieckenst) [reply listener/wallpapers.py:32]: @muse-dev ignore

File: listener/konsolemod/wallpapers.py [hunk 3]
```diff
@@ -5,36 +5,37 @@
...
 
     @wallpaper.command()
-    async def nature(self, ctx):
+    async def nature(self, ctx: SlashContext):
         imgnat = random.choice(desNature.images)
         embednat = discord.Embed(title="Nature Wallpapers", color=0x00ff00, url=imgnat)
         embednat.set_image(url=imgnat)
         await ctx.send(embed=embednat)
     
     @wallpaper.command()
```
[the remark this answers is not in the document]
Comment (pieckenst) [reply listener/konsolemod/wallpapers.py:32]: @muse-dev ignore

File: listener/konsolemod/helper.py [hunk 4]
```diff
@@ -1,82 +1,173 @@
...
+            title="Help System",
+        )    
+        sudodo.add_field(
+            name="**Sudo**",
+            value=help.sudo,
+            inline=False
+        )
+        await ctx.send(embed=sudodo)
     
     @help.command(pass_context=True)
```
[the remark this answers is not in the document]
Comment (pieckenst) [reply listener/konsolemod/helper.py:138]: @muse-dev ignore

File: listener/music/music.py [hunk 5]
```diff
@@ -1,240 +1,734 @@
...
+
+class IncorrectChannelError(commands.CommandError):
+    """Error raised when commands are issued outside of the players session channel."""
+    pass
+
+
+class Track(wavelink.Track):
+    """Wavelink Track object with a requester attribute."""
+
+    __slots__ = ('requester', )
```
[the remark this answers is not in the document]
Comment (pieckenst) [reply listener/music/music.py:32]: @muse-dev ignore

File: listener/music/music.py [hunk 6]
```diff
@@ -1,240 +1,734 @@
...
+
+        elif not await self.is_position_fresh():
+            try:
+                await self.controller.message.delete()
+            except discord.HTTPException:
+                pass
+
+            self.controller.stop()
+
+            self.controller = InteractiveController(embed=self.build_embed(), player=self)
```
[the remark this answers is not in the document]
Comment (pieckenst) [reply listener/music/music.py:110]: @muse-dev ignore

**Comment by fucreview** [comment]
Yes yes i approve

**Review by fucreview — APPROVED** [review]
All fine

**Pull request merged**